contentcuration/contentcuration/utils/assessment/qti/fields.py

```
@@ -0,0 +1,108 @@
+    """Validate and normalize BCP47 language tag."""
+    if not isinstance(value, str):
+        raise ValueError(f"BCP47 language tag must be a string, got {type(value)}")
+
+    if not value:
+        raise ValueError("BCP47 language tag cannot be empty")
+
+    try:
+        # Validate and normalize using langcodes
+        return LangCodesLanguage.get(value).to_tag()
+    except ValueError as e:
+        raise ValueError(f"Invalid BCP47 language tag: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)